server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorFactory.java

```diff
@@ -114,91 +269,39 @@ private static boolean isAggregationSort(BucketOrder order) {
 
     @Override
     protected Aggregator doCreateInternal(ValuesSource valuesSource,
-                                            SearchContext searchContext,
-                                            Aggregator parent,
-                                            boolean collectsFromSingleBucket,
-                                            List<PipelineAggregator> pipelineAggregators,
-                                            Map<String, Object> metaData) throws IOException {
+                                          SearchContext searchContext,
+                                          Aggregator parent,
+                                          boolean collectsFromSingleBucket,
+                                          List<PipelineAggregator> pipelineAggregators,
+                                          Map<String, Object> metaData) throws IOException {
         if (collectsFromSingleBucket == false) {
             return asMultiBucketAggregator(this, searchContext, parent);
         }
+
+        AggregatorSupplier aggregatorSupplier = ValuesSourceRegistry.getInstance().getAggregator(config.valueSourceType(),
```

Comment:
Oh hmm good point.  Should this code be using the passed in valueSourceType instead of the `config` entirely?

Perhaps we make the superclass `config` private so it isn't accessible?  I did a quick skim and it looks like most (all?) uses of `config` are just to get the formatter, so perhaps we add the formatter as a protected variable in ValuesSourceAggregatorFactory instead?

Reply:
Yeah, I think that's the way to go.  I'll add a line item on the meta PR, and one of us can grab it soon.